articles/130_ros_time.md

```diff
@@ -39,85 +45,110 @@ Slower than real time simulation is necessary for complicated systems where accu
-If `/clock` is being published, calls to the ROS time abstraction will return the latest time received from the `/clock` topic.
-If time has not been set it will return zero if nothing has been received.
-A time value of zero should be considered an error meaning that time is uninitialized.
+### ROS Time Message Sources
 
-If the time on the clock jumps backwards, a callback handler will be invoked and be required to complete before any calls to the ROS time abstraction report the new time.
-Calls that come in before that must block.
-The developer has the opportunity to register callbacks with the handler to clear any state from their system if necessary before time will be in the past.
+- An included (where?) Steady Clock Publisher.
+- An included (where?) System Clock Publisher.
+- An included (where?) Gazebo Plugin.
+- The ROSBag playback engine.
 
-The frequency of publishing the `/clock` as well as the granularity are not specified as they are application specific.
+#### Default Time Source
 
-##### No Advanced Estimating Clock
+Users can call `Node::now()` for the current time.
+With the assumption of a single Node instance per process, you can also get the current time from a static method: `Time::now()`.
+Nodes will automatically subscribe for the `/clock` message unless it is programmatically disabled. (Do we need to disable it? We can simply "not publish" to use the system clock.)
+If a publisher exists for the `/clock` topic, it will override the system time when using the ROS time abstraction.
+The `ROSTime` and the `now()` methods will fall back to `SystemTime` (and then `SteadyTime`) when an ROS Time Source is not active (aka, it has been disabled purposely or hasn't received a clock broadcast).
 
-There are more advanced techniques which could be included to attempt to estimate the propagation properties and extrapolate between time ticks.
-However all of these techniques will require making assumptions about the future behavior of the time abstraction.
-And in the case that playback or simulation is instantaneously paused, it will break any of these assumptions.
-There are techniques which would allow potential interpolation, however to make these possible it would require providing guarantees about the continuity of time into the future.
-For more accuracy the progress of time can be slowed, or the frequency of publishing can be increased.
-Tuning the parameters for the `/clock` topic lets you trade off time for computational effort and/or bandwidth.
+Users can also add callbacks to `Node::time_reversing` and `Node::time_jumping_back`.
+The developer has the opportunity to register callbacks with the handler to clear any state from their system if necessary before time will be in the past.
+If the time on the clock jumps backwards, a callback handler will be invoked and be required to complete before any calls to the ROS time abstraction report the new time.
```

Comment:
The purpose of `time_reversing` was for special handling when the real time factor went negative (aka, time was going backwards). However, it sounds like the real time factor complicates too many scenarios; we aren't going with that plan.